### PR TITLE
Feat auth UI and functionality

### PR DESCRIPTION
**Pull request by aishwarya-bm**
1. added UI & functionality for signup/login using firebase
2. added landing page

### CODE REVIEW

File: src/utils/auth.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      const user = userCredential.user;
+      const { accessToken, uid } = user;
+      localStorage.setItem("accessToken", accessToken);
+      dispatch(login(uid));
+      navigate("/home");
+      Toast({ message: "Login successful.", type: "success" });
+    })
+    .catch(error => {
+      const errorCode = error.code;
+      console.log(errorCode);
```
Comment (dev-enforced) [inline src/utils/auth.js]: Can remove this debugger.

File: src/utils/auth.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+         });
+      });
+}
+
+const getLoggedInUserData = async (uid,dispatch, setUserProfile) => {
+  const db = getFirestore(app);
+  const docRef = doc(db, "user_profile", uid);
+  const docSnap = await getDoc(docRef);
+  if (docSnap.exists()) {
+    console.log("Document data:", docSnap.data());
```
Comment (dev-enforced) [inline src/utils/auth.js]: Similarly here.

**Review by dev-enforced — APPROVED** [review]
The code structure looks good. The firebase authentication related changes done in `utils/auth.js` 
was new for me in terms of learning !! Setting up the styling parts of the page using Material Design was also clearly demonstrated while going through the PR. Good work @aishwarya-bm on both the functionality and styling. Hope to see the live preview of **Ssup**  soon to test out this feature!!

**Review by nishasen — APPROVED** [review]
Clean code, like how you are using mui, learnt a lot of new things related to designing

File: src/utils/auth.js (hunk 1 above)
Comment (aishwarya-bm) [reply src/utils/auth.js]: Thanks for the feedback! Also made the changes.

**Pull request merged**